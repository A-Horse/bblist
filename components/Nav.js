import React, { Component, PropTypes } from 'react'
import fetch from 'isomorphic-fetch'
import { connect } from 'react-redux'
import { Link, browserHistory } from 'react-router'

import {makeGravatarHash} from '../services/gravatar';

import {authUser} from '../actions/login';

class Nav extends Component {
  constructor() {
    super()
  }

  componentWillMount() {
    let {dispatch} = this.props;

    
    dispatch(authUser());
  }
  
  render() {
    
    const userCell = this.renderUserCell();
    
    
    return (
      <header>
        <span>Links:</span>
        {' '}
        <Link to="/">Home2</Link>
        {' '}
        <Link to="/foo">Foo</Link>
        {' '}
        <Link to="/bar">Bar</Link>
        {' '}
        { userCell }
      </header>
    )
  }

  renderUserCell() {
    const user = this.props.user;
    
    const cachedUsername = localStorage.getItem('cachedUsername');

    if( cachedUsername ){
      return (
        <Link to="/profile">{cachedUsername}</Link>
      );
    }
    return (
      <div>
        <Link to="/login">Login</Link>
        <Link to="/signup">Sign up</Link>
      </div>
    );
  }
  
  
}

const mapStateToProps = (state) => {
  return {
    user: state.login.state.loginUser
  }
};

export default connect(mapStateToProps)(Nav);

