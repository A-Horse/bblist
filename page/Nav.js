import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { makeGravatarUrl } from 'services/gravatar';
import { Storage, storageImage } from 'services/storage';
import { DropList } from 'components/widget/DropList';
import ClickOutSide from 'components/utils/ClickOutSide';

import { LogoBan } from 'components/commons/LogoBan';

export const navHeight = 42;

import 'style/nav.scss';

class Nav extends Component {
  constructor() {
    super();
    this.state = { dropDownToggle: false };
  }

  componentDidUpdate() {
    if (this.props.user && !Storage.get('avator')) {
      this.refs.avator.onload = () => {
        storageImage('avator', this.refs.avator);
      };
    }
  }

  render() {
    return (
      <nav className="nav">
        <LogoBan white={true} />

        <div className="nav-link-area">
          <Link to="/home">Dash</Link>
          <Link to="/task-wall">Task</Link>
          <Link to="/todo">Todo</Link>
          <Link to="/wiki">Wiki</Link>
          <Link to="/mind">Mind</Link>
        </div>
        {this.renderNavUser()}
      </nav>
    );
  }

  renderNavUser() {
    if (this.props.userIsFetching) {
      // TODO: 移到下面去
      return null;
    }
    console.log(this.props);

    const userName = this.props.user.username;
    return (
      <ClickOutSide
        className="avatar-area"
        onClickOutside={() => {
          this.state.dropDownToggle && this.setState({ dropDownToggle: false });
        }}
      >
        {this.renderAvatar()}
        <DropList toggle={this.state.dropDownToggle}>
          <ul>
            <li>
              Signed in as <strong>{userName}</strong>
            </li>
            <li>
              <Link to="/profile">Profile</Link>
            </li>
            <li>
              <Link to="/setting">setting</Link>
            </li>
            <li onClick={this.props.actions.logout}>logout</li>
          </ul>
        </DropList>
      </ClickOutSide>
    );
  }

  renderAvatar() {
    const { user } = this.props;
    const avatarData = Storage.get('avator');
    return !!avatarData
      ? <img
          ref="avator"
          className="nav-avatar"
          src={`data:image/png;base64,${avatarData}`}
          onClick={() => this.setState({ dropDownToggle: !this.state.dropDownToggle })}
        />
      : <img
          ref="avator"
          className="nav-avatar"
          crossOrigin="Anonymous"
          src={makeGravatarUrl(user.email)}
          onClick={() => this.setState({ dropDownToggle: !this.state.dropDownToggle })}
        />;
  }
}

export default Nav;
