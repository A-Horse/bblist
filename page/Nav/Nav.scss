@import 'style/variable';

nav.nav {
  position: relative;
  width: 100%;
  z-index: 100;
  height: $nav-height;
  background-color: $red;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 20px;
  box-sizing: border-box;

  .logoban {
    flex-shrink: 2;
    color: black;
    height: 24px;
    margin-top: -1px;

    .logo {
      border-radius: $radius-small;
      height: 24px;
    }

    .text-logo {
      margin-left: 8px;
      height: 16px;
    }

    @media screen and (max-width: 600px) {
      .text-logo {
        display: none;
      }
    }
  }

  .nav-link-area {
    flex-grow: 2;
    margin-top: 3px;

    .nav-links__large-device {
      margin-top: 8px;
      margin-left: 20px;

      ul {
        float: left;

        li {
          float: left;

          &.active {
            font-weight: 900;
            text-shadow: 0 0 1px #fff;
          }
        }

        a {
          margin-left: 10px;
          color: #fff;
          letter-spacing: 0.3px;
        }
      }
    }

    .nav-links__small-device {

    }

    .nav-active-link__small-device {
      text-align: center;
      cursor: pointer;
      user-select: none;
      .fa {
        margin-left: 8px;
        vertical-align: middle;
        color: $white;
      }

      ul {
        display: inline-block;
      }

      li {
        display: none;

        &.active {
          display: block;
          font-weight: bold;

          a {
            color: $white;
          }
        }
      }
    }
  }

  .avatar-area {
    position: relative;
    line-height: $nav-height;
    height: 100%;
  }

  .nav-avatar {
    margin-top: -1px;
    height: 27px;
    width: 27px;
    border-radius: 50%;
    vertical-align: middle;
    cursor: pointer;
  }

  .drop-list {
    width: 180px;
    border-radius: $radius-middle;
    box-shadow: 0 0 3px #eee;
     li {
       padding: 0 10px;
       font-size: $font-size-sm;
       color: $text-primary;
       cursor: pointer;
       border-bottom: 1px solid $color-divider;
       text-overflow: ellipsis;
       overflow: hidden;
       white-space: nowrap;

       &:last-child {
         border-bottom: none;
       }
       a {
         color: inherit;
         display: inline-block;
         height: 100%;
         width: 100%;
       }
    }
  }
}
