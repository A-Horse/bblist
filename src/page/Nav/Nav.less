@import "../../style/variable.less";

.app-header {
  background-color: @primary-color !important;
  padding: 0 20px !important;
  letter-spacing: 1px;

  .app-header-menu {
    line-height: 46px;
    padding-left: 30px;
    display: flex;
    justify-content: space-between;

    a {
      color: #f8f8f8;
      text-decoration: none;
      margin-left: 10px;
      display: inline-block;
      width: 80px;
    }
    a.active {
      color: #fff;
      font-weight: 500;
    }
  }

  .nav-avatar {
    width: 30px;
    height: 30px;
    border-radius: 5px;
    cursor: pointer;
  }

  .logoban {
    float: left;
    display: inline-block;
    margin-top: -3px;
  }
  .logo {
    width: 20px;
    height: 20px;
  }

  .logo-text {
    width: 80px;
    margin: 8px 0 0 3px;
  }
}
