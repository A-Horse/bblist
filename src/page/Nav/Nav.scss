.AppNav {
  display: flex;
  align-items: center;
  background-color: #4697f5;
  height: 38px;
  padding: 0 12px;
  position: relative;
  z-index: 10;

  &--ban {
    color: white;
    align-items: center;
    display: flex;
    flex: 0;

    .Logo {
      border-radius: 50%;
    }

    .TextLogo {
      color: white;
      margin-left: 4px;
    }
  }

  .AppNav--menu {
    padding-left: 12px;
    display: flex;
    justify-content: space-between;
    flex-grow: 100;

    .app-header-menu--link-container {
      display: flex;
      justify-content: center;
      align-items: center;
    }

    a {
      color: #777;
      text-decoration: none;
      margin-left: 10px;
      display: inline-block;
      width: 60px;
      height: 26px;
      line-height: 26px;
      font-weight: 500;
      letter-spacing: 0.5px;
      text-align: center;
      border-radius: 6px;
      background-color: white;
    }

    a.active {
      background-color: white;
      color: #6b69e0;
    }
  }

  .AppNav--Avatar {
    display: flex;
    align-items: center;
    flex: 0;

    .AppButton {
      padding: 0;
      height: 32px;

      img {
        background-color: white;
        border-radius: 50% !important;
      }

      &:hover {
        background-color: white;
      }
    }
  }
}
