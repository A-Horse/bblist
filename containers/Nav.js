import { connect } from 'react-redux';
import Nav from 'components/Nav';
import { authUser } from 'actions/login';
import { removeCachedData } from 'utils/auth';
import { wrapDispathToAction } from 'utils/wrap-props';
import { logout } from 'actions/logout';
import { browserHistory } from 'react-router';

const actions = {};

const mapDispatchToProps = dispatch => {
  return {
    logout() {
      return dispatch(logout()).then(() => {
        browserHistory.push('/signin');
      });
    }
  };
};

const mapStateToProps = state => {
  return {
    userIsFetching: state.auth.isFetching,
    user: state.auth.loginedUser
    // path: state.router.location.pathname
  };
};

const NavContainer = connect(mapStateToProps, wrapDispathToAction(actions, mapDispatchToProps))(
  Nav
);

export default NavContainer;
