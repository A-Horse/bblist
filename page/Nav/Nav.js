import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { makeGravatarUrl } from 'services/gravatar';
import { Storage, storageImage } from 'services/storage';
import { DropList } from 'components/widget/DropList';
import ClickOutSide from 'components/utils/ClickOutSide';
import { LogoBan } from 'components/commons/LogoBan';
import { activeClassWhenMatchPrefix } from '../../utils/route';

export const navHeight = 42;

import './Nav.scss';

class Nav extends Component {
  constructor() {
    super();
    this.state = { dropDownToggle: false };
  }

  componentDidUpdate() {
    if (this.props.user && !Storage.get('avator')) {
      this.refs.avator.onload = () => {
        storageImage('avator', this.refs.avator);
      };
    }
  }

  render() {
    return (
      <nav className="nav">
        <LogoBan white={true} />

        <div className="nav-link-area">
          <Link to="/home">Dash</Link>
          <Link to="/task-wall">Task</Link>
          <Link to="/todo" className={activeClassWhenMatchPrefix('/todo')}>
            Todo
          </Link>
          <Link to="/wiki">Wiki</Link>
          <Link to="/mind">Mind</Link>
        </div>
        {this.renderNavUser()}
      </nav>
    );
  }

  renderNavUser() {
    if (this.props.identifyFetching) {
      return null;
    }
    const userName = this.props.user.get('username');
    return (
      <ClickOutSide
        className="avatar-area"
        onClickOutside={() => {
          this.state.dropDownToggle && this.setState({ dropDownToggle: false });
        }}
      >
        <span>
          {userName}
        </span>
        {this.renderAvatar()}
        <DropList toggle={this.state.dropDownToggle}>
          <ul>
            <li>
              Signed in as <strong>{userName}</strong>
            </li>
            <li>
              <Link to="/profile">Profile</Link>
            </li>
            <li>
              <Link to="/setting">setting</Link>
            </li>
            <li className="logout-button" onClick={this.props.actions.LOGOUT_FN}>
              logout
            </li>
          </ul>
        </DropList>
      </ClickOutSide>
    );
  }

  renderAvatar() {
    const { user } = this.props;
    const avatarData = Storage.get('avator');
    return !!avatarData
      ? <img
          ref="avator"
          className="nav-avatar"
          src={`data:image/png;base64,${avatarData}`}
          onClick={() => this.setState({ dropDownToggle: !this.state.dropDownToggle })}
        />
      : <img
          ref="avator"
          className="nav-avatar"
          crossOrigin="Anonymous"
          src={makeGravatarUrl(user.get('email'))}
          onClick={() => this.setState({ dropDownToggle: !this.state.dropDownToggle })}
        />;
  }
}

export default Nav;
