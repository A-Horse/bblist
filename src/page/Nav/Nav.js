//
import React, { Component } from 'react';
import { Link, NavLink } from 'react-router-dom';
import { makeGravatarUrl } from '../../services/gravatar';
import { Storage, storageImage } from '../../services/storage';
import { LogoBan } from '../../components/commons/LogoBan/LogoBan';

import { Layout, Icon, Dropdown, Menu } from 'antd';
const { Header } = Layout;

// TODO remove
export const navHeight = 42;

import './Nav.less';

class Nav extends Component {
  state = {};

  links = [
    { name: 'Dash', url: '/home' },
    { name: 'Task', url: '/task-board' },
    { name: 'Todo', url: '/todo' }
  ];

  onNewAvatarImageLoaded = event => {
    storageImage('avator', event.nativeEvent.target);
  };

  render() {
    const userName = this.props.user.username;
    const { user } = this.props;
    const avatarData = Storage.get('avator');

    const menu = (
      <Menu>
        <Menu.Item>
          <Link to="/profile">
            Signed in as <strong>{userName}</strong>
          </Link>
        </Menu.Item>
        <Menu.Item>
          <Link to="/setting">Setting</Link>
        </Menu.Item>

        <Menu.Item>
          <span onClick={this.props.actions.LOGOUT_REQUEST}>Logout</span>
        </Menu.Item>
      </Menu>
    );
    return (
      <Header className="app-header">
        <Link to="/task-board">
          <LogoBan white={true} />
        </Link>
        <div className="app-header-menu">
          <div>
            <NavLink to="/task-board" activeClassName="active">
              <Icon type="appstore-o" />
              Project
            </NavLink>

            <NavLink to="/todo" activeClassName="active">
              <Icon type="check-square" />
              Todo
            </NavLink>
          </div>

          <div style={{ display: 'inline-block' }}>
            <Dropdown overlay={menu} placement="bottomRight">
              {avatarData ? (
                <img
                  ref={ref => (this.avator = ref)}
                  className="nav-avatar"
                  src={`data:image/png;base64,${avatarData}`}
                />
              ) : (
                <img
                  ref={ref => (this.avator = ref)}
                  className="nav-avatar"
                  crossOrigin="Anonymous"
                  onLoad={this.onNewAvatarImageLoaded}
                  src={makeGravatarUrl(user.email)}
                />
              )}
            </Dropdown>
          </div>
        </div>
      </Header>
    );
  }
}

export default Nav;
