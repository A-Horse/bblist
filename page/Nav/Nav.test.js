import React from 'react';
import Nav from './Nav';
import renderer from 'react-test-renderer';
// import ShallowRenderer from 'react-test-renderer/shallow';
import { MemoryRouter } from 'react-router';
import { fromJS } from 'immutable';
import history from '../../services/history';
import { mount } from 'enzyme';

const user = fromJS({
  created_at: null,
  desc: null,
  email: 'jest@facebook.com',
  id: 34,
  status: null,
  type: null,
  updated_at: null,
  username: 'jest'
});

test('Website Nav', () => {
  const actions = {
    LOGOUT_FN: jest.fn()
  };
  const tree = renderer.create(
    <MemoryRouter>
      <Nav user={user} actions={actions} />
    </MemoryRouter>
  );
  expect(tree).toMatchSnapshot();
});

test('Website Nav behavior', () => {
  history.location.pathname = '/todo';

  const actions = {
    LOGOUT_FN: jest.fn()
  };
  const wrapper = mount(
    <MemoryRouter>
      <Nav user={user} actions={actions} />
    </MemoryRouter>
  );

  expect(wrapper.find('.nav-link-area .active').text()).toEqual('Todo');
  wrapper.find('.nav-avatar').simulate('click');
  wrapper.find('.logout-button').simulate('click');
  expect(actions.LOGOUT_FN.mock.calls.length).toBe(1);
});
