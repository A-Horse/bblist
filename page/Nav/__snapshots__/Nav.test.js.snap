// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Website Nav 1`] = `
<nav
  className="nav"
>
  <div
    className="logoban"
    onClick={[Function]}
  >
    <img
      className="logo"
      src="/static/logo.png"
    />
    <img
      className="text-logo"
      src="/static/text-logo-white.svg"
    />
  </div>
  <div
    className="nav-link-area"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      Dash
    </a>
    <a
      href="/task-wall"
      onClick={[Function]}
    >
      Task
    </a>
    <a
      className=""
      href="/todo"
      onClick={[Function]}
    >
      Todo
    </a>
    <a
      href="/wiki"
      onClick={[Function]}
    >
      Wiki
    </a>
    <a
      href="/mind"
      onClick={[Function]}
    >
      Mind
    </a>
  </div>
  <div
    className="avatar-area"
  >
    <span>
      jest
    </span>
    <img
      className="nav-avatar"
      crossOrigin="Anonymous"
      onClick={[Function]}
      src="https://www.gravatar.com/avatar/50fe9076fe9e2143e68bd431c75ba1da"
    />
    <noscript />
  </div>
</nav>
`;
